extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/internal/XMLResponseProcessor.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    // initializes our XML parsing state machine
+    private void init() {
+        stateSwitchingMap = new HashMap<>();
+        Map<String, XMLHandlerState> msgBodyMap = new HashMap<>();
+        stateSwitchingMap.put(XMLHandlerState.MsgBody, msgBodyMap);
+        msgBodyMap.put("info", XMLHandlerState.Info);
+        msgBodyMap.put("volume", XMLHandlerState.Volume);
+        msgBodyMap.put("presets", XMLHandlerState.Presets);
+        msgBodyMap.put("key", XMLHandlerState.Unprocessed); // only confirmation of our key presses...
+        msgBodyMap.put("status", XMLHandlerState.Unprocessed); // only confirmation of commands sent to device...
+        msgBodyMap.put("zone", XMLHandlerState.Zone); // only confirmation of our key presses...
+        msgBodyMap.put("bass", XMLHandlerState.Bass);
+        msgBodyMap.put("sources", XMLHandlerState.Sources);
+        msgBodyMap.put("bassCapabilities", XMLHandlerState.BassCapabilities);
+
+        // info message states
+        Map<String, XMLHandlerState> infoMap = new HashMap<>();
+        stateSwitchingMap.put(XMLHandlerState.Info, infoMap);
+        infoMap.put("components", XMLHandlerState.Unprocessed); // TODO read software version and serial number
+        infoMap.put("name", XMLHandlerState.InfoName);
+        infoMap.put("type", XMLHandlerState.InfoType);
+        infoMap.put("networkInfo", XMLHandlerState.Unprocessed);
+        infoMap.put("margeAccountUUID", XMLHandlerState.Unprocessed);
+        infoMap.put("margeURL", XMLHandlerState.Unprocessed);
+        infoMap.put("moduleType", XMLHandlerState.Unprocessed);
+        infoMap.put("variant", XMLHandlerState.Unprocessed);
+        infoMap.put("variantMode", XMLHandlerState.Unprocessed);
+        infoMap.put("countryCode", XMLHandlerState.Unprocessed);
+        infoMap.put("regionCode", XMLHandlerState.Unprocessed);
+
+        Map<String, XMLHandlerState> updatesMap = new HashMap<>();
+        stateSwitchingMap.put(XMLHandlerState.Updates, updatesMap);
+        updatesMap.put("clockDisplayUpdated", XMLHandlerState.Unprocessed); // can we get anything useful of that?
+        updatesMap.put("connectionStateUpdated", XMLHandlerState.UnprocessedNoTextExpected);
+        updatesMap.put("infoUpdated", XMLHandlerState.Unprocessed);
+        updatesMap.put("nowPlayingUpdated", XMLHandlerState.MsgBody);
+        updatesMap.put("nowSelectionUpdated", XMLHandlerState.Unprocessed); // TODO this seems to be quite a useful info
+                                                                            // what is currently played..
+        updatesMap.put("recentsUpdated", XMLHandlerState.Unprocessed);
+        updatesMap.put("volumeUpdated", XMLHandlerState.MsgBody);
+        updatesMap.put("zoneUpdated", XMLHandlerState.ZoneUpdated); // just notifies but dosn't provide details
+        updatesMap.put("bassUpdated", XMLHandlerState.BassUpdated);
+        updatesMap.put("volumeUpdated", XMLHandlerState.VolumeUpdated);

```

Comment:
This seems the point with `volumeUpdated`. It is put here into the map two times - one for MsgBody in line 87 (that would work for the message as we see it here) and one time for 'VolumeUpdated' in line 90 overriding the value from line 87. I traced the `VolumeUpdated` part and it seems in the past only the `<volumeUpdated/>` tag without the `<volume>` data was sent and so `<volumeUpdated/>` triggered a fetch for the current volume. Seeing the message from the log it seems we could clean up the entire `VolumeUpdated` path...